cmd/kubectl-k8ssandra/register/command.go

```diff
@@ -0,0 +1,73 @@
+package register
+
+import (
+	"fmt"
+
+	"github.com/charmbracelet/log"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+var RegisterClusterCmd = &cobra.Command{
+	Use:   "register [flags]",
+	Short: "register a data plane into the control plane.",
+	Long:  `register creates a ServiceAccount on a source cluster, copies its credentials and then creates a secret containing them on the destination cluster. It then also creates a ClientConfig on the destination cluster to reference the secret.`,
+	Run:   entrypoint,
+}
+
+func SetupRegisterClusterCmd(cmd *cobra.Command, streams genericclioptions.IOStreams) {
+	RegisterClusterCmd.Flags().String("source-kubeconfig",
+		"",
+		"path to source cluster's kubeconfig file - defaults to KUBECONFIG then ~/.kube/config")
+	RegisterClusterCmd.Flags().String("dest-kubeconfig",
+		"",
+		"path to destination cluster's kubeconfig file - defaults to KUBECONFIG then ~/.kube/config")
+	RegisterClusterCmd.Flags().String("source-context", "", "context name for source cluster")
+	RegisterClusterCmd.Flags().String("dest-context", "", "context name for destination cluster")
+	RegisterClusterCmd.Flags().String("source-namespace", "k8ssandra-operator", "namespace containing service account for source cluster")
+	RegisterClusterCmd.Flags().String("dest-namespace", "k8ssandra-operator", "namespace where secret and clientConfig will be created on destination cluster")
+	RegisterClusterCmd.Flags().String("serviceaccount-name", "k8ssandra-operator", "serviceaccount name for destination cluster")
+	RegisterClusterCmd.Flags().String("destination-name", "remote-k8ssandra-operator", "name for remote clientConfig and secret on destination cluster")
```

Comment:
**Issue:** from a usability standpoint I think this is going to cause frustration for our users. Could we use a sanitized version of the source context name for this instead of a hardcoded value?
This would bring us on par with the current behavior of the registration script and will also make it much easier to reason about once we list the available contexts.
There's also the issue that if you try to register two clusters consecutively, the second one will fail after a lot of retries, complaining about : `Registration continuing error creating secret. err: secrets "remote-k8ssandra-operator" already exists sa k8ssandra-operator`

Since `remote-k8ssandra-operator` wasn't something the user provided in the first place, it will create some confusion.

Reply:
> Issue: from a usability standpoint I think this is going to cause frustration for our users. Could we use a sanitized version of the source context name for this instead of a hardcoded value?

I don't understand this question, these are not hardcoded, they are defaults.

Reply:
> There's also the issue that if you try to register two clusters consecutively, the second one will fail after a lot of retries, complaining about : Registration continuing error creating secret. err: secrets "remote-k8ssandra-operator" already exists sa k8ssandra-operator

This is going to be the case no matter what. We can't do this dynamically within the Cobra framework as defaults are registered at init, not when user flags values are available. 

We could write defaulting code, but I think at that point things just get more confusing as magic is happening behind the scenes.

If usability here is a concern I recommend we enhance the help message.

Reply:
My suggestion would rather be to leave the default empty and use the sanitized source context name as destination name, unless the flag is used to pass an explicit value.

Reply:
Done.